akka-http-caching/src/main/scala/akka/http/caching/LfuCache.scala

```diff
@@ -105,7 +105,21 @@ private[caching] class LfuCache[K, V](val store: AsyncCache[K, V]) extends Cache
 
   def apply(key: K, genValue: () => Future[V]): Future[V] = store.get(key, toJavaMappingFunction[K, V](genValue)).toScala
 
-  def getOrLoad(key: K, loadValue: K => Future[V]) = store.get(key, toJavaMappingFunction[K, V](loadValue)).toScala
+  def put(key: K, mayBeValue: Future[V])(implicit ex: ExecutionContext): Future[V] = {
+    val previouslyCacheValue = Option(store.getIfPresent(key))
+
+    previouslyCacheValue match {
+      case None =>
+        store.put(key, toJava(mayBeValue).toCompletableFuture)
+        mayBeValue
+      case _ => mayBeValue.map { value =>
```

Comment:
```suggestion
      case _ => mayBeValue.foreach { value =>
```

Reply:
(this does `map` so the `mayBeValue` is returned to the user after the `store.put` was invoked. I agree it'd be OK to do `foreach` here and just return `mayBeValue` directly, no preference)

Reply:
Ah, indeed, then `map` sounds fine to me as well.